lib/rules/no-timers.js

```diff
@@ -11,8 +28,9 @@ module.exports = {
   create(context) {
     return {
       CallExpression(node) {
-        let name = node.callee.name || node.callee.property.name;
-        if (isTimer(name)) {
+        let name = getName(node);
+
+        if (name && isTimer(name) && !isInAction(node) && !isGuarded(node)) {
```

Comment:
You could use `setTimeout` without a guard on server side because it is exposed. 

Nitpick: You could use window/document API for hooks that don't run on server side : `didInsertElement`, `willInsertElement`. So this will give a false positive in those cases.